### PR TITLE
Fixing monitor_install_path_base

### PR DESCRIPTION
**Pull request by bentsi**

### CODE REVIEW

**Comment by fgelcer** [comment]
have succeeded on my local test after this change.

File: sdcm/cluster.py [hunk 1]
```diff
@@ -2967,7 +2967,7 @@ def __init__(self, targets, params):
         self.sct_ip_port = self.set_local_sct_ip()
         self.grafana_port = 3000
         self.monitor_branch = self.params.get('monitor_branch', default='branch-2.3')
-        self.monitor_install_path_base = "/home/{}".format(self.params.get('ami_monitor_user'), default='centos')
+        self.monitor_install_path_base = "~"
```
Comment (aleksbykov) [inline sdcm/cluster.py]: this cause exception if files were copied via scp
Command: 'scp -r -o StrictHostKeyChecking=no -o BatchMode=yes -o ConnectTimeout=300 -o ServerAliveInterval=300 -o UserKnownHostsFile=/var/tmp/tmpEwMudZ -P 22 -i /home/abykov/.ssh/scylla-qa-ec2 /var/tmp/tmpFhuyC4/prometheus.yml.template \'centos@3.93.183.21:"~/scylla-grafana-monitoring-branch-2.3/prometheus/prometheus.yml.template"\''

Comment (aleksbykov) [reply sdcm/cluster.py]: if 'ami_monitor_user' in self.params:
            self.monitor_install_path_base = "/home/{}".format(self.params.get('ami_monitor_user'), default='centos')
        else:
            self.monitor_install_path_base = "~"

possible fix

Comment (bentsi) [reply sdcm/cluster.py]: I think more robust way is to get the current user HOME directory and store it as a class attribute, then use it

Comment (fgelcer) [reply sdcm/cluster.py]: echo $HOME returns you the home directory for the current user

**Comment by fruch** [comment]
Since we can't call the any nodes in the `__init__` making them lazy, and cache the `$HOME` value
(I'll open it as a PR tomorrow morning)

```diff
diff --git a/sdcm/cluster.py b/sdcm/cluster.py
index 6158b5a..3f81b8f 100644
--- a/sdcm/cluster.py
+++ b/sdcm/cluster.py
@@ -3033,13 +3033,28 @@ class BaseMonitorSet(object):
         self.sct_ip_port = self.set_local_sct_ip()
         self.grafana_port = 3000
         self.monitor_branch = self.params.get('monitor_branch', default='branch-2.3')
-        self.monitor_install_path_base = "/home/{}".format(self.params.get('ami_monitor_user'), default='centos')
-        self.monitor_install_path = os.path.join(self.monitor_install_path_base,
-                                                 "scylla-grafana-monitoring-{}".format(self.monitor_branch))
-        self.monitoring_conf_dir = os.path.join(self.monitor_install_path, "config")
-        self.monitoring_data_dir = os.path.join(self.monitor_install_path_base, "scylla-grafana-monitoring-data")
+        self._monitor_install_path_base = None
         self.phantomjs_installed = False
 
+    @property
+    def monitor_install_path_base(self):
+        if not self._monitor_install_path_base:
+            self._monitor_install_path_base = self.nodes[0].remoter.run("echo $HOME").stdout
+        return self._monitor_install_path_base
+
+    @property
+    def monitor_install_path(self)
+        return os.path.join(self.monitor_install_path_base, "scylla-grafana-monitoring-{}".format(self.monitor_branch))
+
+    @property
+    def monitoring_conf_dir(self):
+        return os.path.join(self.monitor_install_path, "config")
+
+    @property
+    def monitoring_data_dir(self):
+        return os.path.join(self.monitor_install_path_base, "scylla-grafana-monitoring-data")
+
+
     @staticmethod
     def get_node_ips_param(public_ip=True):
         param_name = 'monitor_nodes_public_ip' if public_ip else 'monitor_nodes_private_ip'

```

**Pull request closed without merging**